core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -2850,11 +2839,16 @@ static RexNode fix(RexBuilder rexBuilder, RexNode term) {
 
     @SuppressWarnings({"rawtypes", "unchecked", "UnstableApiUsage"})
     <C extends Comparable<C>> Sarg<C> build(boolean negate) {
+      final RexUnknownAs unknownAs =
+          nullAsTrueCount > 0 ? TRUE
+              : nullAsUnknownCount > 0 ? UNKNOWN
+                  : nullAsFalseCount > 0 ? FALSE
```

Comment:
Very confusing code, does these three flags have priority difference ?

Reply:
The priority difference is inherent in how three-valued logic works. Consider: TRUE OR UNKNOWN OR FALSE returns TRUE; UNKNOWN OR FALSE returns UNKNOWN.

Reply:
Thanks for the explanation ~

Reply:
I'll add explanatory comments.